File: 1206-corporate-flight-bookings/1206-corporate-flight-bookings.py
class Solution:
    def corpFlightBookings(self, bookings: List[List[int]], n: int) -> List[int]:
        flights = []
        number_of_seats = [0] * (n + 1) 

        for booking in bookings:
            number_of_seats[booking[0] - 1] += booking[2]
            number_of_seats[booking[1]] -= booking[2]

        for i in range(1, n):
            number_of_seats[i] += number_of_seats[i - 1]

        return number_of_seats[:-1]
        

File: 1206-corporate-flight-bookings/1206-corporate-flight-bookings.go
func corpFlightBookings(bookings [][]int, n int) []int {
    numberOfSeats := make([]int, n + 1)

    for _, booking := range bookings {
        numberOfSeats[booking[0] - 1] += booking[2]
        numberOfSeats[booking[1]] -= booking[2]
    }

    for i := 1; i < n; i++ {
        numberOfSeats[i] += numberOfSeats[i - 1]
    }

    return numberOfSeats[:n]
}